pkg/operator/ceph/disruption/clusterdisruption/static_pdb.go

```diff
@@ -20,28 +20,40 @@ import (
 	"context"
 
 	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	policyv1 "k8s.io/api/policy/v1"
 	policyv1beta1 "k8s.io/api/policy/v1beta1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 )
 
-func (r *ReconcileClusterDisruption) createStaticPDB(pdb *policyv1beta1.PodDisruptionBudget) error {
+func (r *ReconcileClusterDisruption) createStaticPDB(pdb client.Object) error {
 	err := r.client.Create(context.TODO(), pdb)
 	if err != nil {
-		return errors.Wrapf(err, "failed to create pdb %q", pdb.Name)
+		return errors.Wrapf(err, "failed to create pdb %q", pdb.GetName())
 	}
 	return nil
 }
 
-func (r *ReconcileClusterDisruption) reconcileStaticPDB(request types.NamespacedName, pdb *policyv1beta1.PodDisruptionBudget) error {
-	existingPDB := &policyv1beta1.PodDisruptionBudget{}
-	err := r.client.Get(context.TODO(), request, existingPDB)
+func (r *ReconcileClusterDisruption) reconcileStaticPDB(request types.NamespacedName, pdb client.Object) error {
+	var existingPDB client.Object
+	usePDBV1, err := k8sutil.UsePDBV1Version(r.context.ClusterdContext.Clientset)
+	if err != nil {
+		return err
```

Comment:
Wrap your errors with meaningful messages.

Reply:
I am returning this ` errors.Wrapf(err, "failed to fetch k8s version") `  wrap error which is present in [ k8utils  ](https://github.com/rook/rook/pull/7977/files#diff-c815b31fa3ee2cc6a06fe74fa48e47bcb9f37c88718447373e886f65406a8a76) did I need to wrap it again?

Reply:
Yes.  Its easier to debug when we wrap errors with meaningful messages where ever possible.

Reply:
Done, have a look :)